### PR TITLE
[FSTORE-626] boilerplate fm persistence

### PR DESCRIPTION
**Pull request by vatj**
This PR adds/fixes/changes...
- please summarize your changes to the code 
- and make sure to include all changes to user-facing APIs

JIRA Issue: -

Priority for Review: -

Related PRs: -

**How Has This Been Tested?**

- [x] Unit Tests
- [x] Integration Tests
- [x] Manual Tests on VM

**Checklist For The Assigned Reviewer:**

```
- [ ] Checked if merge conflicts with master exist
- [ ] Checked if stylechecks for Java and Python pass
- [ ] Checked if all docstrings were added and/or updated appropriately
- [ ] Ran spellcheck on docstring
- [ ] Checked if guides & concepts need to be updated
- [ ] Checked if naming conventions for parameters and variables were followed
- [ ] Checked if private methods are properly declared and used
- [ ] Checked if hard-to-understand areas of code are commented
- [ ] Checked if tests are effective
- [ ] Built and deployed changes on dev VM and tested manually
- [x] (Checked if all type annotations were added and/or updated appropriately)
```

### CODE REVIEW

File: python/hsfs/feature_group.py [hunk 1]
```diff
@@ -46,6 +46,8 @@
 from hsfs.core.job import Job
 from hsfs.core.variable_api import VariableApi
 from hsfs.core import great_expectation_engine
+from hsfs.core.feature_monitoring_config import FeatureMonitoringConfig
+from hsfs.core.feature_monitoring_config_engine import FeatureMonitoringConfigEngine
 
 
 class FeatureGroupBase:
```
Comment (javierdlrm) [inline python/hsfs/feature_group.py:53]: I guess the `enable_feature_monitoring` method is missing (calling `enable_feature_monitoring_config`in the engine class)

File: python/hsfs/feature_group.py [hunk 2]
```diff
@@ -965,6 +970,29 @@ def get_validation_history(
...
 
+    def enabled_statistics_monitoring(
+        self,
+        feature_name: str,
+        window_configuration_type: str,
+        time_offset: str,
+        window_length: str,
+        scheduler_config: str,
+    ) -> FeatureMonitoringConfig:
+        detection_monitoring_window_configuration = (
```
Comment (javierdlrm) [inline python/hsfs/feature_group.py:981]: Just an opinion, I think we can use the abbreviation `config` everywhere to keep it consistent and shorter. (as we have it already in multiple places)

File: python/hsfs/feature_group.py [hunk 3]
```diff
@@ -965,6 +970,29 @@ def get_validation_history(
                 "Only Feature Group registered with Hopsworks can fetch validation history."
             )
 
+    def enabled_statistics_monitoring(
+        self,
+        feature_name: str,
+        window_configuration_type: str,
+        time_offset: str,
+        window_length: str,
+        scheduler_config: str,
```
Comment (javierdlrm) [inline python/hsfs/feature_group.py:979]: `scheduler_config: str = None`?

Comment (javierdlrm) [inline python/hsfs/feature_group.py:979]: Parameter `row_percentage` is missing in `enabled_statistics_monitoring.`

Also, `time_offset`, `window_length` and `row_percentage` should be nullable I believe.

File: python/hsfs/core/feature_monitoring_result.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+import humps
+from typing import Any, Dict, List, Optional
+from hsfs import util
+
+
+class FeatureMonitoringResult:
+    def __init__(
+        self,
+        feature_store_id: int,
+        entity_id: int,
```
Comment (javierdlrm) [inline python/hsfs/core/feature_monitoring_result.py:27]: I think we can skip `entity_id`. Please see my [comment here](https://github.com/logicalclocks/hopsworks-ee/pull/1267/files#r1115999003).

File: python/hsfs/feature_group.py [hunk 5]
```diff
@@ -965,6 +970,29 @@ def get_validation_history(
                 "Only Feature Group registered with Hopsworks can fetch validation history."
             )
 
+    def enabled_statistics_monitoring(
```
[the remark this answers is not in the document]
Comment (vatj) [reply python/hsfs/feature_group.py:973]: Ja, I think there is more work to be done on the user facing api so I can just remove that method for now

**Pull request merged**